lib/internal/bootstrap/node.js

```diff
@@ -663,9 +663,42 @@
     const test = Function.call.bind(RegExp.prototype.test);
 
     const {
-      allowedV8EnvironmentFlags,
-      allowedNodeEnvironmentFlags
-    } = process.binding('config');
+      getOptions,
+      types: { kV8Option },
+      envSettings: { kAllowedInEnvironment }
+    } = internalBinding('options');
+    const { options, aliases } = getOptions();
+    const allowedV8EnvironmentFlags =
+      [...options].filter(([name, info]) =>
+        info.type === kV8Option &&
+          info.envVarSettings === kAllowedInEnvironment)
+      .map(([name]) => name);
```

Comment:
I would prefer to pass through filter options into `getOptions` instead of doing the filtering here.
Since it has a string option at the moment, it would likely be best to accept an object for the options instead.

Another alternative to the current way would be to only go over the entries once (this is also nicer to read for me):

```js
const allowedV8EnvironmentFlags = [];
const allowedNodeEnvironmentFlags = [];
for (const [name, info] of options) {
  if (info.envVarSettings === kAllowedInEnvironment) {
    if (info.type === KV8Option) {
      allowedV8EnvironmentFlags.push(name);
    } else {
      allowedNodeEnvironmentFlags.push(name);
    }
  }
}
```